### PR TITLE
#970 -- Added new custom events for filter-search bar component

### PR DESCRIPTION
**Pull request by paboden**
New events are triggered when a filter select option is toggled on or off, when a chip is removed, and when the filtering is cleared/reset.

New event names are 'filterSearchMultiSelectToggle', 'filterSearchChipRemoval', and 'filterSearchClear'

@Jura is this in the ballpark of what we talked about. These are events on the filter/search element that people can use in their own scripts.

### CODE REVIEW

File: stories/assets/js/filter-search-bar.js [hunk 1]
```diff
@@ -73,15 +95,37 @@ const toggleFilter = function () {
...
   jQuery(document).on('click', '.sort-filter-search', function () {
-    jQuery(this).toggleClass('close');
-    jQuery(this).next('.search-filter').toggleClass('show-filter');
+    const $el = jQuery(this);
+    $el.toggleClass('close');
+    $el.next('.search-filter').toggleClass('show-filter');
+  });
+
+  // For testing custom events
+  jQuery(document).on('filterSearchChipRemoval', (event) => {
```
[the remark this answers is not in the document]
Comment (paboden) [reply stories/assets/js/filter-search-bar.js]: The testing code section has been removed.

**Comment by paboden** [comment]
Hi @Jura I moved the custom event for the multi-select/select toggling and the custom event for the checkbox toggling into their respective multi-select.js and select.js files. I've updated the filter-search.js to use those new custom events. I've also tested those events locally in other scripts with successful triggering. 
I haven't updated the documentation yet.
Let me know if we are getting closer to desired setup.
Cheers
Phil

**Review by Jura — APPROVED** [review]
Looking good, please proceed with documenting your changes so we can wrap it up

**Comment by Jura** [comment]
@paboden , thanks for the documentation, it is pretty what we needed. I used it to create a mini-app to test functionality and accuracy of instructions and there are a few issues:
- [ ] select.min.js is part of this PR, but not referenced in documentation. Should it be loaded to make widget function properly or not? If so - please update documentation, if not - why it is a part of this PR?
- [ ] There is a typo in JS reference: https://undp.github.io/design-system/js/filter-seach-bar.min.js (missing 'r' in '-search-'
- [ ] multiSelectToggle event data is not providing reference to "checkbox_id" property as it is stated in documentation. I would also ask to use "select_id" property name for that to avoid confusion 

https://codepen.io/jurakhrapunov/pen/eYPKGLO

**Review by Jura — APPROVED** [review]

**Pull request merged**